cloud/services/loadbalancers/loadbalancers.go

```diff
@@ -191,25 +191,29 @@ func (s *Service) Delete(ctx context.Context) error {
 
 // getAvailablePrivateIP checks if the desired private IP address is available in a virtual network.
 // If the IP address is taken or empty, it will make an attempt to find an available IP in the same subnet
-func (s *Service) getAvailablePrivateIP(ctx context.Context, resourceGroup, vnetName, subnetCIDR, PreferredIPAddress string) (string, error) {
+func (s *Service) getAvailablePrivateIP(ctx context.Context, resourceGroup, vnetName, PreferredIPAddress string, subnetCIDRs []string, isIPv6 bool) (string, error) {
 	ip := PreferredIPAddress
 	if ip == "" {
 		ip = azure.DefaultInternalLBIPAddress
+		subnetCIDR := subnetCIDRs[0]
```

Comment:
is it guaranteed at this point that the slice will have length of at least 1?

How do we select ipv4 vs ipv6 here? This would rely on the user to specify the one they want in the correct order (the one they want first in the list).

Reply:
the a good point, I guess technically it's not (even though we always set it). To be honest this whole logic is fragile, I just set it like that thinking "I'll deal with this later".